### PR TITLE
tools/dm: add doc for query status

### PR DESCRIPTION
**Pull request by dcalvin**
Added doc for DM Query Status
Translated from: https://github.com/pingcap/docs/blob/master/tools/dm/query-status.md

### CODE REVIEW

File: tools/dm/query-status.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                                "unsynced": [                                   # 未执行该 sharding DDL 语句的上游表。如有上游表未完成同步，`blockingDDLs` 为空。
+                                ]
+                            }
+                        ],
+                        "synced": false         # 增量同步是否已和上游一致，并与上游数据库拥有相同的 binlog 位置。由于后台 `Sync` 单元并不会实时刷新保存点，当前值为 "false"
+                                                # 并不一定代表发生了同步延迟。
+                    }
+                }
+            ],
+            "relayStatus": {    # The synchronization status of the relay log.
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]:
```suggestion
            "relayStatus": {    # relay 单元的同步状态.
```

File: tools/dm/query-status.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                "masterBinlog": "(bin.000001, 3234)",                               # 上游数据库的 binlog 位置。
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]: ditto

File: tools/dm/query-status.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                "masterBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-14",    # 上游数据库的 binlog GTID 信息。
+                "relaySubDir": "c0149e17-dff1-11e8-b6a8-0242ac110004.000001",       # 当前使用的 relay log 子目录。
+                "relayBinlog": "(bin.000001, 3234)",                                # 已被拉取至本地存储的 binlog 位置。
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]: ditto

File: tools/dm/query-status.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                "relayBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-14",     # 已被拉取至本地存储的 binlog GTID 信息。
+                "relayCatchUpMaster": true,     # 本地 relay log 同步进度是否与上游一致。
+                "stage": "Running",             # relay log 的 `Sync` 处理单元状态
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]:
```suggestion
                "stage": "Running",             # relay 处理单元状态
```

File: tools/dm/query-status.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                "masterBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
+                "relaySubDir": "c0149e17-dff1-11e8-b6a8-0242ac110004.000001",
+                "relayBinlog": "(bin.000001, 28507)",
+                "relayBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
+                "relayCatchUpMaster": true,
+                "stage": "Running",
+                "result": null
+            }
+        },
+        {
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]: we can remove L101 ~ L129, duplicated with L72 ~ L100.
original CN doc's fault.

File: tools/dm/query-status.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                "masterBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
+                "relaySubDir": "c0149e17-dff1-11e8-b6a8-0242ac110004.000001",
+                "relayBinlog": "(bin.000001, 28507)",
+                "relayBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
+                "relayCatchUpMaster": true,
+                "stage": "Running",
+                "result": null
+            }
+        }
+                {
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]:
```suggestion
        {
```

File: tools/dm/query-status.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+- `Stopped`: 
+
+    - 停止状态。
+    - 如在子任务为 `Running` 或 `Paused` 状态下执行 `stop-task` 命令，任务状态会切换为 `Stopped`。
+    - 如子任务处于该状态，不可使用 `resume-task` 命令恢复任务。
+
+- `Finished`: 
+
+    - 任务完成状态。
+    - 只有完整同步任务正常完成后，任务才会切换为该状态。
```
Comment (csuzhangxc) [inline tools/dm/query-status.md]:
```suggestion
    - 只有 `task-mode` 为 `full` 的任务正常完成后，任务才会切换为该状态。
```

File: tools/dm/query-status.md [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+            "msg": "",
+            "subTaskStatus": [
+                {
+                    "name": "test",
+                    "stage": "Running",
+                    "unit": "Load",
+                    "result": null,
+                    "unresolvedDDLLockID": "",
+                    "load": {                   # `Load` 处理单元的同步信息。
+                        "finishedBytes": "115", # 已加载字节数。
```
Comment (amyangfei) [inline tools/dm/query-status.md]:
```suggestion
                        "finishedBytes": "115", # 已全量导入字节数。
```

File: tools/dm/query-status.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+                        "totalBytes": "452",    # 待加载字节数。
```
Comment (amyangfei) [inline tools/dm/query-status.md]:
```suggestion
                        "totalBytes": "452",    # 总计需要导入的字节数。
```

File: tools/dm/query-status.md [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+                        "progress": "25.44 %"   # 加载进度。
```
Comment (amyangfei) [inline tools/dm/query-status.md]:
```suggestion
                        "progress": "25.44 %"   # 全量导入进度。
```

File: tools/dm/query-status.md [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+    - 停止状态。
+    - 如在子任务为 `Running` 或 `Paused` 状态下执行 `stop-task` 命令，任务状态会切换为 `Stopped`。
+    - 如子任务处于该状态，不可使用 `resume-task` 命令恢复任务。
+
+- `Finished`: 
+
+    - 任务完成状态。
+    - 只有完整同步任务正常完成后，任务才会切换为该状态。
+
+### 状态转换图表
```
Comment (amyangfei) [inline tools/dm/query-status.md]:
```suggestion
### 状态转换图
```

File: tools/dm/query-status.md [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+                    "unit": "Sync",         # DM 的处理单元，包括 ”Check“，"Dump“，”Load“，以及”Sync“。
+                    "result": null,         # 子任务失败时显示错误信息。
+                    "unresolvedDDLLockID": "test-`test`.`t_target`",    # sharding DDL lock ID，用于异常情况下手动处理 sharding DDL lock。
+                    "sync": {                   # 同一组件下，当前 `Sync` 处理单元的同步信息。 
+                        "totalEvents": "12",    # 该子任务中同步的 binlog event 总数。
+                        "totalTps": "1",        # 该子任务中每秒同步的 binlog event 数量。
+                        "recentTps": "1",       # 该子任务中最后一秒同步的 binlog event 数量。
+                        "masterBinlog": "(bin.000001, 3234)",                               # binlog 在上游数据库所处位置。
+                        "masterBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-14",    # 上游数据库的 GTID 信息。
+                        "syncerBinlog": "(bin.000001, 2525)",                               # 已被 `Sync` 处理单元同步的 binlog 位置。
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply tools/dm/query-status.md]: Got it.

File: tools/dm/query-status.md (hunk 2 above)
Comment (dcalvin) [reply tools/dm/query-status.md]: Already in the same line

File: tools/dm/query-status.md (hunk 3 above)
Comment (dcalvin) [reply tools/dm/query-status.md]: Already in the same line

File: tools/dm/query-status.md [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+                    "name": "test",         # 子任务名称。
+                    "stage": "Running",     # 子任务运行状态，包括 “New”，“Running“，”Paused“，“Stopped’，以及 ”Finished“。
+                    "unit": "Sync",         # DM 的处理单元，包括 ”Check“，"Dump“，”Load“，以及”Sync“。
+                    "result": null,         # 子任务失败时显示错误信息。
+                    "unresolvedDDLLockID": "test-`test`.`t_target`",    # sharding DDL lock ID，用于异常情况下手动处理 sharding DDL lock。
+                    "sync": {                   # 同一组件下，当前 `Sync` 处理单元的同步信息。 
+                        "totalEvents": "12",    # 该子任务中同步的 binlog event 总数。
+                        "totalTps": "1",        # 该子任务中每秒同步的 binlog event 数量。
+                        "recentTps": "1",       # 该子任务中最后一秒同步的 binlog event 数量。
+                        "masterBinlog": "(bin.000001, 3234)",                               # binlog 在上游数据库所处位置。
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply tools/dm/query-status.md]: @dcalvin what's your opinion?

[the remark this answers is not in the document]
Comment (dcalvin) [reply tools/dm/query-status.md]: I agree with your suggested translation. It's technically more accurate.

**Comment by dcalvin** [comment]
Thanks @amyangfei @csuzhangxc PTAL again.

**Comment by csuzhangxc** [comment]
@dcalvin seems still there are some quotes not matched in
```
子任务运行状态，包括 “New”，“Running“，”Paused“，“Stopped“ 以及 ”Finished“。
```
and
```
DM 的处理单元，包括 ”Check“，“Dump“，”Load“ 以及 ”Sync“。
```

**Comment by dcalvin** [comment]
> ”



> @dcalvin seems still there are some quotes not matched in
> 
> ```
> 子任务运行状态，包括 “New”，“Running“，”Paused“，“Stopped“ 以及 ”Finished“。
> ```
> and
> 
> ```
> DM 的处理单元，包括 ”Check“，“Dump“，”Load“ 以及 ”Sync“。
> ```



> @dcalvin seems still there are some quotes not matched in
> 
> ```
> 子任务运行状态，包括 “New”，“Running“，”Paused“，“Stopped“ 以及 ”Finished“。
> ```
> and
> 
> ```
> DM 的处理单元，包括 ”Check“，“Dump“，”Load“ 以及 ”Sync“。
> ```

On second thought, I suggest we remove these quotation marks. The facts that they are English in Chinese, and their initial letters are capitalized already are sufficient to distinguish them as statuses. Quotation mark always makes texts less tidy. What do you think? @lilin90

**Comment by dcalvin** [comment]
> Please add a link to "query-status" in "可通过 query-status 获得" of the `break-ddl-lock` section in the tools/dm/manually-handling-sharding-ddl-locks.md file.

done

**Pull request merged**